sympy/physics/mechanics/functions.py

```diff
@@ -532,7 +539,7 @@ def _recurser(expr, sub_dict):
 
 def _fraction_decomp(expr):
     """Return num, den such that expr = num/den"""
-    if not isinstance(expr, Mul):
+    if not isinstance(expr, MulClass):
```

Comment:
So I take it C++ doesn't let you dispatch to a different type in the constructor like Python does. 

It should be possible to use a single name `Add` in the Python wrapper. You just need a class `Add` whose `__new__` calls your `add` function and whose `__instancecheck__` returns `True` for subclasses of the actual wrapper `Add` class. This will break if any code in SymPy uses `type(x) is Add` instead of `isinstance(x, Add)`, but [it shouldn't be doing that anyway](https://github.com/sympy/sympy/issues/6751).